lib/galaxy/tools/parameters/basic.py

```diff
@@ -235,6 +234,17 @@ def build( cls, tool, param ):
         else:
             return parameter_types[param_type]( tool, param )
 
+    @classmethod
+    def parse_name(cls, input_source):
+        name = input_source.get("name", None)
+        if name is None:
+            argument = input_source.get("argument")
+            if argument:
+                name = argument.lstrip("-")
+            else:
```

Comment:
@jmchilton this means that you can create a `param` without a `name` and it will take the stripped `argument`?
I don't think we should allow this. `name` should be required, `argument` and be optional and needs to be unstripped (for UI and gsh.)